sympy/sets/fancysets.py

```diff
@@ -443,3 +445,451 @@ def _sup(self):
     @property
     def _boundary(self):
         return self
+
+
+def normalize_theta_set(theta):
+    """
+    Normalize a Real Set theta in the Interval [0, 2*pi). It currently
+    supports Interval and FiniteSet. It Returns a the normalized value
+    of theta in the Set. For Interval, a maximum of one cycle [0, 2*pi],
+    is returned i.e. for theta equal to [0, 10*pi], returned normalized
+    value would be [0, 2*pi). As of now it supports theta as FiniteSet
+    and Interval.
+
+    Raises
+    ======
+
+    NotImplementedError
+        The algorithms for Normalizing theta Set are not yet
+        implemented.
+    ValueError
+        The input is not valid, i.e. the input is not a real set.
+    RuntimeError
+        It is a bug, please report to the github issue tracker.
+
+    Examples
+    ========
+
+    >>> from sympy.sets.fancysets import normalize_theta_set
+    >>> from sympy import Interval, FiniteSet, pi
+    >>> normalize_theta_set(Interval(9*pi/2, 5*pi))
+    [pi/2, pi]
+    >>> normalize_theta_set(Interval(-3*pi/2, pi/2))
+    [0, 2*pi)
+    >>> normalize_theta_set(Interval(-pi/2, pi/2))
+    [0, pi/2] U [3*pi/2, 2*pi)
+    >>> normalize_theta_set(Interval(-4*pi, 3*pi))
+    [0, 2*pi)
+    >>> normalize_theta_set(Interval(-3*pi/2, -pi/2))
+    [pi/2, 3*pi/2]
+    >>> normalize_theta_set(FiniteSet(0, pi, 3*pi))
+    {0, pi}
+
+    """
+    from sympy.functions.elementary.trigonometric import _pi_coeff as coeff
+    from sympy.functions.elementary.complexes import Abs
+
+    if theta.is_Interval:
+        if Abs(theta.args[0] - theta.args[1]) >= 2*S.Pi:
+            return Interval(0, 2*S.Pi, False, True)
+
+        new_theta = []
+        for val in [theta.args[0], theta.args[1]]:
+            k = coeff(val)
+            if (not k) and (k != S.Zero):
+                raise NotImplementedError('Normalizing theta without pi as'
+                                          'coefficient, is not Implemented.')
+            elif k == S.Zero:
+                if val == S.Zero:
+                    new_theta.append(S.Zero)
+                else:
+                    # when theta is n*pi
+                    new_theta.append(2*S.Pi)
+            else:
+                new_theta.append(k*S.Pi)
+        # one complete circle
+        if (new_theta[0] == new_theta[1]):
+            return Interval(0, 2*S.Pi, False, True)
```

Comment:
I just noticed, these two lines aren't needed anymore, since we already covered the case of complete circle above in [Line 493](https://github.com/sympy/sympy/pull/9463/files#diff-c1c17bdd28aafd400bf9415c7d048382R493). I would remove these.